packages/target-ethers-v5/src/codegen/types.ts

```diff
@@ -141,10 +141,22 @@ export function generateOutputComplexTypesAsObject(
   return namedElementsCode
 }
 
-function generateArrayOrTupleType(item: string, length?: number) {
-  if (length !== undefined && length < 6) {
-    return `[${Array(length).fill(item).join(', ')}]`
-  } else {
-    return `(${item})[]`
+function generateArrayOrTupleType(item: string, originalType: string): string {
```

Comment:
I think all of this should be moved to `parseEvmType` directly in TypeChain core.